kernel_issues/__init__.py

```diff
@@ -0,0 +1,47 @@
+import logging
+import re
+from squad.plugins import Plugin as BasePlugin
+
+
+logger = logging.getLogger()
+
+# Tip: broader regexes should come first
+REGEXES = [
+    ('kernel_panic', 'Kernel panic - not syncing.*?end Kernel panic - not syncing'),
+    ('kernel_exception', '------------\[ cut here \]------------.*?------------\[ cut here \]------------'),
+    ('kernel_trace', 'Stack:.*?---\[ end trace \w* \]---'),
+    ('kernel_oops', 'Oops:.*?$'),
+    ('kernel_fault', 'Unhandled fault.*?$'),
+    ('kernel_warning','WARNING:.*?$'),
+]
+
+
+class KernelIssues(BasePlugin):
+    name = "Kernel Issues"    
+
+    def __prepare_regexes(self):
+        combined = [r'(%s)' % r[1] for r in REGEXES]
+        return re.compile(r'|'.join(combined), re.S | re.M)
+
+    def __kernel_msgs_only(self, log):
+        kernel_msgs = re.findall(r'^\[[ \d]+\.[ \d]+\] .*?$', log, re.S | re.M)
+        return '\n'.join(kernel_msgs)
+
+    def postprocess_testrun(self, testrun):
+        log = self.__kernel_msgs_only(testrun.log)
+        regex = self.__prepare_regexes()
+        matches = regex.findall(log)
+        snippets = {regex_id: [] for regex_id in range(len(REGEXES))}
+        for match in matches:
+            for regex_id in range(len(REGEXES)):
+                if len(match[regex_id]) > 0:
+                    snippets[regex_id].append(match[regex_id])
+
+        suite, _ = testrun.build.project.suites.get_or_create(slug='kernel-issues')
+        for regex_id in range(len(REGEXES)):
+            testrun.tests.create(
+                suite=suite,
+                name=REGEXES[regex_id][0],
```

Comment:
Can you give an example of what suite and name will resolve to?

Reply:
kernel-issues/kernel_panic
kernel-issues/kernel_exception
kernel-issues/kernel_trace
kernel-issues/kernel_oops
kernel-issues/kernel_fault
kernel-issues/kernel_warning

these names are defined at the top

Reply:
We run ~20 lava jobs per environment, and so this test result will then get overwritten 20 times and last one wins. The name needs to be per-job somehow.